docs/tutorials/quick-start/quick-start-part-1.md

```diff
@@ -0,0 +1,1107 @@
+
+So, instead of this:
+
+```js
+function handwrittenReducer(state, action) {
+  return {
+    ...state,
+    first: {
+      ...state.first,
+      second: {
+        ...state.first.second,
+        [action.someId]: {
+          ...state.first.second[action.someId],
+          fourth: action.someValue
+        }
+      }
+    }
+  }
+}
+```
+
+You can write code that looks like this:
+
+```js
+function reducerWithImmer(state, action) {
+  state.first.second[action.someId].fourth = action.someValue
+}
+```
+
+That's a lot easier to read!
+
+But, here's something _very_ important to remember:
+
+:::warning
+
+**You can _only_ write "mutating" logic in Redux Toolkit's `createSlice` and `createReducer` because they use Immer inside! If you write mutating logic in reducers without Immer, it _will_ mutate the state and cause bugs!**
+
+:::
+
+With that in mind, let's go back and look at the actual reducers from the counter slice.
+
+```js title="features/counter/counterSlice.js"
+export const counterSlice = createSlice({
+  name: 'counter',
+  initialState: {
+    value: 0
+  },
+  reducers: {
+    increment: state => {
+      // Redux Toolkit allows us to write "mutating" logic in reducers. It
+      // doesn't actually mutate the state because it uses the immer library,
+      // which detects changes to a "draft state" and produces a brand new
+      // immutable state based off those changes
+      state.value += 1
+    },
+    decrement: state => {
+      state.value -= 1
+    },
+    incrementByAmount: (state, action) => {
+      // highlight-next-line
+      state.value += action.payload
+    }
+  }
+})
+```
+
+We can see that the `increment` reducer will always add 1 to `state.value`. Because Immer knows we've made changes to the draft `state` object, we don't have to actually return anything here. In the same way, the `decrement` reducer subtracts 1.
+
+In both of those reducers, we don't actually need to have our code look at the `action` object. It will be passed in anyway, but since we don't need it, we can skip declaring `action` as a parameter for the reducers.
+
+On the other hand, the `incrementByAmount` reducer _does_ need to know something: how much it should be adding to the counter value. So, we declare the reducer as having both `state` and `action` arguments. In this case, we know that the amount we typed into the textbox is being put into the `action.payload` field, so we can add that to `state.value`.
+
+:::info Want to Know More?
+
+For more information on immutability and writing immutable updates, see [the "Immutable Update Patterns" docs page](../../recipes/structuring-reducers/ImmutableUpdatePatterns.md) and [The Complete Guide to Immutability in React and Redux](https://daveceddia.com/react-redux-immutability-guide/).
+
+:::
+
+### Writing Async Logic with Thunks
+
+The next function that's exported from `counterSlice` might look a bit strange:
```

Comment:
I would add an introduction to this. I would want to know what you mean by "writing async logic with thunks", but it just dives right into some code. Perhaps add a small explanation here, like "JavaScript applications often need to make asynchronous API calls to retrieve data. Redux Thunks are required to do this." or whatever.